src/library_async.js

```diff
@@ -573,6 +573,34 @@ mergeInto(LibraryManager.library, {
     afterUnwind: null,
     asyncFinalizers: [], // functions to run when *all* asynchronicity is done
 
+#if ASSERTIONS
+    instrumentWasmImports: function(imports) {
+      var ASYNCIFY_IMPORTS = {{{ JSON.stringify(ASYNCIFY_IMPORTS) }}};
+      for (var x in imports) {
+        (function(x) {
```

Comment:
Any reason you use a IIFE here?   wouldn't this be basically the same with line 580 and 599 removed?

Reply:
Sadly it's necessary... The closure lower down (line 583) captures the outside environment, and x is updated in that environment. So we would get the final value of x when that closure is eventually called, and not the current x which is what we want.

This isn't an issue in most languages because block scope creates a new x, but JS has no block scope (not without `let` at least).